podcasts/Onboarding/Plus/UpgradeLandingView.swift

```diff
@@ -0,0 +1,324 @@
+import SwiftUI
+
+struct UpgradeLandingView: View {
+    @EnvironmentObject var viewModel: PlusLandingViewModel
+
+    private let tiers: [UpgradeTier] = [.plus, .patron]
+
+    @State private var currentPage: Int = 0
+
+    @State private var displayPrice: DisplayPrice = .yearly
+
+    var body: some View {
+        VStack(spacing: 0) {
+            topBar
+
+            ZStack {
+
+                if currentPage == 0 {
+                    tiers[0].background
+                        .transition(.opacity.animation(currentPage == 0 ? .easeIn : .easeOut))
+                        .ignoresSafeArea()
+                } else {
+                    tiers[1].background
+                        .transition(.opacity.animation(currentPage == 0 ? .easeIn : .easeOut))
+                        .ignoresSafeArea()
+                }
```

Comment:
Wdyt about making this more dynamic to reduce the code duplication?

```suggestion
                let selectedTier = tiers[currentPage]

                ForEach(tiers) { tier in
                    tier.background
                        .opacity(selectedTier.id == tier.id ? 1 : 0)
                        .ignoresSafeArea()
                }
```

Reply:
Done in 20d385879c39bb02a9565f49a527b590674fecc9